src/gax-internal/src/observability/grpc_tracing.rs

```diff
@@ -25,6 +25,10 @@ use tower::{Layer, Service};
 #[derive(Clone, Copy, Debug)]
 pub struct AttemptCount(pub i64);
 
+/// A wrapper for the resource name to be stored in request extensions.
+#[derive(Clone, Debug)]
+pub struct ResourceName(pub String);
```

Comment:
This is a nit:


```suggestion
pub struct ResourceName(String);

impl ResourceName {
  pub fn new(value: String) -> Self { Self(value) }
  pub fn as_str(&self) -> &str { self.0.as_str() }
}
```

I prefer wrapper types to hide things...